### PR TITLE
ccip integration test fixes

### PR DESCRIPTION
**Pull request by dimkouv**
Several fixes in the integration test setup

### CODE REVIEW

File: integration-tests/deployment/address_book.go [hunk 1]
```diff
@@ -163,3 +163,19 @@ func NewMemoryAddressBookFromMap(addressesByChain map[uint64]map[string]TypeAndV
 		AddressesByChain: addressesByChain,
 	}
 }
+
+// SearchAddressBook search an address book for a given chain and contract type and return the first matching address.
+func SearchAddressBook(ab AddressBook, chain uint64, typ ContractType) (string, error) {
```
Comment (makramkd) [inline integration-tests/deployment/address_book.go:168]: Can be method on address book?

File: integration-tests/deployment/ccip/changeset/initial_deploy.go [hunk 2]
```diff
@@ -12,8 +12,7 @@ import (
 // TODO: Maybe there's a generics approach here?
 // Note if the change set is a deployment and it fails we have 2 options:
 // - Just throw away the addresses, fix issue and try again (potentially expensive on mainnet)
-func InitialDeployChangeSet(env deployment.Environment, c ccipdeployment.DeployCCIPContractConfig) (deployment.ChangesetOutput, error) {
-	ab := deployment.NewMemoryAddressBook()
+func InitialDeployChangeSet(ab deployment.AddressBook, env deployment.Environment, c ccipdeployment.DeployCCIPContractConfig) (deployment.ChangesetOutput, error) {
 	err := ccipdeployment.DeployCCIPContracts(env, ab, c)
```
Comment (makramkd) [inline integration-tests/deployment/ccip/changeset/initial_deploy.go:16]: cc @connorwstein

File: integration-tests/deployment/ccip/deploy_home_chain.go [hunk 3]
```diff
@@ -375,23 +376,13 @@ func BuildAddDONArgs(
 			OfframpAddress:        offRamp.Address().Bytes(),
```
Comment (makramkd) [inline integration-tests/deployment/ccip/deploy_home_chain.go:376]: Not related to this PR but we `common.LeftPadBytes` of the onramp address on the offramp, should this also be `common.LeftPadBytes`'d?

File: integration-tests/deployment/ccip/deploy_home_chain.go [hunk 4]
```diff
@@ -375,23 +376,13 @@ func BuildAddDONArgs(
...
 			Nodes:                 ocrNodes,
 			OffchainConfig:        offchainConfig,
-			// TODO: Deploy RMNHome and set address here
-			RmnHomeAddress: common.BytesToAddress(randomBytes(20)).Bytes(),
+			RmnHomeAddress:        rmnHomeAddress,
```
Comment (makramkd) [inline integration-tests/deployment/ccip/deploy_home_chain.go:379]: Similar q as the above on this?

Comment (dimkouv) [reply integration-tests/deployment/ccip/deploy_home_chain.go:379]: I assume it's ok, since the test would fail earlier otherwise

**Review by 0xnogo — APPROVED** [review]

**Review by skudasov — APPROVED** [review]

**Review by lukaszcl — APPROVED** [review]
the test code looks good!

**Pull request merged**